test/parallel_api/iterator/permutation_iterator.pass.cpp

```diff
@@ -39,15 +155,557 @@ main()
+{
+    DEFINE_TEST_PERM_IT_CONSTRUCTOR(test_transform_reduce)
+
+    template <typename TIterator>
+    void generate_data(TIterator itBegin, TIterator itEnd, TestValueType initVal)
+    {
+        ::std::iota(itBegin, itEnd, initVal);
+    }
+
+    template <typename Policy, typename Iterator1, typename Size>
+    void
+    operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Size n)
+    {
+        if constexpr (can_run_nonmodify_test<Iterator1>())
+        {
+            TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);     // source data for transform_reduce
+
+            test_through_permutation_iterator_mibt<Iterator1, Size, PermItIndexTag>{first1, n}(
+                [&](auto permItBegin, auto permItEnd)
+                {
+                    // Fill full source data set (not only values iterated by permutation iterator)
+                    generate_data(host_keys.get(), host_keys.get() + n, TestValueType{});
+                    host_keys.update_data();
+        
+                    const auto result = dpl::transform_reduce(exec, permItBegin, permItEnd, TestValueType{},
+                                                              ::std::plus<TestValueType>(), ::std::negate<TestValueType>());
+                    wait_and_throw(exec);
+
+                    // KSATODO required to implement results check
+                    //const auto expected =
+                    //    ::std::transform_reduce(permItBegin, permItEnd, TestValueType{}, ::std::plus<TestValueType>(),
+                    //                            ::std::negate<TestValueType>());
+                    //
+                    //EXPECT_EQ(expected, result, "Wrong result of dpl::transform_reduce");
+#if ASSERT_ON_NOT_IMPLEMENTED_CHECK
+                    assert(false);
+#endif // ASSERT_ON_NOT_IMPLEMENTED_CHECK
+                });
+        }
+    }
+};
+
+// dpl::merge, dpl::inplace_merge -> __parallel_merge
+DEFINE_TEST_PERM_IT(test_merge, PermItIndexTag)
+{
+    DEFINE_TEST_PERM_IT_CONSTRUCTOR(test_merge)
+
+    template <typename TIterator>
+    void generate_data(TIterator itBegin, TIterator itEnd, TestValueType initVal)
+    {
+        ::std::iota(itBegin, itEnd, initVal);
+    }
+
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
+    void
+    operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Iterator2 last2, Iterator3 first3, Iterator3 last3, Size n)
+    {
+        if constexpr (can_run_nonmodify_test<Iterator1>())
+        {
+            TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);     // source data(1) for merge
+            TestDataTransfer<UDTKind::eVals, Size> host_vals(*this, n);     // source data(2) for merge
+            TestDataTransfer<UDTKind::eRes,  Size> host_res (*this, n);     // merge results
+
+            if (::std::distance(first3, last3) < (::std::distance(first1, last1) + ::std::distance(first2, last2)))
```

Comment:
It is better write as following
`assert(::std::distance(first3, last3) >= (::std::distance(first1, last1) + ::std::distance(first2, last2)))`